components/TextFields/tests/snapshot/MDCBaseTextFieldSnapshotTests.m

```diff
@@ -18,21 +18,50 @@
 
 #import "MaterialTextFields+ContainedInputView.h"
 
+static const NSTimeInterval kTextFieldValidationAnimationTimeout = 1.0;
+
 @interface MDCBaseTextFieldTestsSnapshotTests : MDCSnapshotTestCase
+@property(strong, nonatomic) MDCBaseTextField *textField;
 @end
 
 @implementation MDCBaseTextFieldTestsSnapshotTests
 
 - (void)setUp {
   [super setUp];
 
+  self.textField = [self createBaseTextFieldInKeyWindow];
   // Uncomment below to recreate all the goldens (or add the following line to the specific
   // test you wish to recreate the golden for).
   //    self.recordMode = YES;
 }
 
 - (void)tearDown {
   [super tearDown];
+  [self.textField removeFromSuperview];
+  self.textField = nil;
+}
+
+- (MDCBaseTextField *)createBaseTextFieldInKeyWindow {
+  MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 200, 60)];
+  textField.borderStyle = UITextBorderStyleRoundedRect;
+  UIWindow *keyWindow = [[UIApplication sharedApplication] keyWindow];
+  [keyWindow addSubview:textField];
+  return textField;
+}
+
+- (void)validateTextField:(MDCBaseTextField *)textField {
+  XCTestExpectation *expectation =
+      [[XCTestExpectation alloc] initWithDescription:@"textfield_validation_expectation"];
+  dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Nit: Im worried this dispatch async will cause collisions with other textfield snapshot tests that are also waiting to be first responder. I suppose we just have to keep an eye out for them.

FYI: Here is another textfield run loop oddity
https://github.com/material-components/material-components-ios/blob/b38372192ff5f7d3273c87e901a2eab402cabef3/components/TextFields/tests/unit/TextFieldControllerLegacyTests.swift#L25

waiting for 1 second in a unit test may be expensive though.